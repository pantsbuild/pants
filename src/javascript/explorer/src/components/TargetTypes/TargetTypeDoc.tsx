import { Remark } from 'react-remark'
import Card, { CardProps } from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Divider from '@mui/material/Divider';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import { TargetInfo, TargetFieldInfo } from "lib/target-data/docs";

export const target_types_fields = [
  "alias",
  "provider",
  "summary",
  "description",
  "fields.alias",
  "fields.provider",
  "fields.description",
  "fields.typeHint",
  "fields.required",
  "fields.default",
];

type TargetTypeFieldDocProps = CardProps & {
  info: TargetFieldInfo;
};

export const TargetTypeFieldDoc = ({ info, sx, ...props }: TargetTypeFieldDocProps) => (
  <Card sx={{ mt: 2, mx: 3, ...sx }} {...props}>
    <CardContent>
      <Typography variant="h6"><code>{info.alias}</code>{info.required && " *"}</Typography>
      <Typography variant="subtitle2">{info.provider}</Typography>
      <Divider />
      <Remark>{info.description}</Remark>
    </CardContent>
  </Card>
);

type TargetTypeDocProps = {
  info?: TargetInfo;
};

export const TargetTypeDoc = ({ info }: TargetTypeDocProps) => info ? (
  <>
    <Card>
      <CardContent>
        <Typography variant="h6"><code>{info.alias}</code></Typography>
        <Typography variant="subtitle2">{info.summary}</Typography>
        <Divider />
        <Remark>{info.description}</Remark>
      </CardContent>
    </Card>
    {info.fields.map((field, index) => (
      <TargetTypeFieldDoc key={index} info={field} />
    ))}
  </>
) : null;
