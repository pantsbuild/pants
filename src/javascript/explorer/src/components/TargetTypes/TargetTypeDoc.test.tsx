import { act, default as renderer, ReactTestRenderer } from 'react-test-renderer';
import { TargetTypeDoc } from './TargetTypeDoc';


test('TargetTypeDoc renders help card.', async () => {
  const info = {
    alias: 'target-type',
    provider: 'test',
    summary: 'Describes the `target-type` target.',
    description: 'More help text.',
    fields: [
      {
        alias: 'field_a',
        provider: 'test',
        description: 'Info for `field_a`.',
        type_hint: 'str',
        required: true,
        default: 'foo',
      },
      {
        alias: 'field_b',
        provider: 'plugged',
        description: 'Info for `field_b`.',
        type_hint: 'str',
        required: false,
        default: null,
      },
    ],
  };

  let component
  await act(async () => {
    component = renderer.create(
      <TargetTypeDoc info={info} />
    );
  });

  let tree = (component as unknown as ReactTestRenderer).toJSON();
  expect(tree).toMatchSnapshot();
});
