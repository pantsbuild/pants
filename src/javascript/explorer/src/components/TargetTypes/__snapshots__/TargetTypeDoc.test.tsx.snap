// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetTypeDoc renders help card. 1`] = `
Array [
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        <code>
          target-type
        </code>
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        Describes the \`target-type\` target.
      </h6>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <p>
        More help text.
      </p>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-11wej7y-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        <code>
          field_a
        </code>
         *
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        test
      </h6>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <p>
        Info for 
        <code>
          field_a
        </code>
        .
      </p>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-11wej7y-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        <code>
          field_b
        </code>
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        plugged
      </h6>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <p>
        Info for 
        <code>
          field_b
        </code>
        .
      </p>
    </div>
  </div>,
]
`;
